examples/form_data/form_data.rs

```diff
@@ -14,15 +13,16 @@ fn main() {
     });
 
     server.post("/confirmation", middleware!{ |req, res|
-        let mut form_data = String::new();
-        req.origin.read_to_string(&mut form_data).unwrap();
+        let form_data = try_with!(res, req.form_body());
 
-        println!("{}", form_data);
+        println!("{:?}", form_data);
 
         let mut data = HashMap::new();
         data.insert("title", "Confirmation");
-        data.insert("formData", &form_data);
-
+        data.insert("firstname", form_data.get("firstname").unwrap_or("What's your first name?"));
+        data.insert("lastname", form_data.get("lastname").unwrap_or("What's your last name?"));
+        data.insert("phone", form_data.get("phone").unwrap_or("Give me your phone or I will give you mine!"));
+        data.insert("email", form_data.get("email").unwrap_or("I want your email!"));
```

Comment:
The tests don't really exercise the `unwrap_or`s here. 

Compared to the json api, where you could state your required fields by being non-`Option` types, this api looks like it's going to require a lot of extra validation by the user.

Maybe we can add a method `require_keys()` to this, which can return an error with a list of the missing keys, e.g.:
```rust
if let Err(missing) = form_data.require_keys(["firstname", "lastname", "email"]) {
    // rerender the form with missing fields highlighted
}